src/main/java/hexgui/util/Platform.java

```diff
@@ -0,0 +1,186 @@
+// Platform.java
+
+package hexgui.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.net.InetAddress;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.Locale;
+
+/** Static utility functions for platform detection and platform-dependent behavior. */
+public class Platform {
+  /** Handler for events from the Application Menu on MacOS. */
+  public interface SpecialMacHandler {
+    /**
+     * Handle about menu event.
+     *
+     * @return true if event was handled successfully.
+     */
+    boolean handleAbout();
+
+    /**
+     * Handle open file event.
+     *
+     * @param filename name of file.
+     * @return true if event was handled successfully.
+     */
+    boolean handleOpenFile(String filename);
+
+    /**
+     * Handle quit application event.
+     *
+     * @return true if event was handled successfully, false if quit should be aborted.
+     */
+    boolean handleQuit();
+  }
+
+  public static String getJavaRuntimeName() {
+    // java.runtime.name is not a standard property
+    String name = System.getProperty("java.runtime.name");
+    if (name == null) name = System.getProperty("java.vm.name");
+    return name;
+  }
+
+  /**
+   * Return information on this computer. Returns host name and cpu information (if /proc/cpuinfo
+   * exists).
+   */
+  public static String getHostInfo() {
+    String info;
+    try {
+      info = InetAddress.getLocalHost().getHostName();
+    } catch (UnknownHostException e) {
+      info = "?";
+    }
+    try {
+      if (existsProcCpuinfo()) {
+        String[] cmdArray = {"/bin/sh", "-c", "grep '^model name' /proc/cpuinfo"};
+        String result = ProcessUtil.runCommand(cmdArray);
+        int start = result.indexOf(':');
+        if (start >= 0) {
+          info = info + " (";
+          int end = result.indexOf("\n");
+          if (end >= 0) info = info + result.substring(start + 1, end).trim();
+          else info = info + result.substring(start + 1).trim();
+          info = info + ")";
+        }
+      }
+    } catch (IOException e) {
+    }
+    return info;
+  }
+
+  /** Check if the platform is Mac OS X. */
+  public static boolean isMac() {
+    return s_isMac;
+  }
+
+  /** Check if the platform is Unix. */
+  public static boolean isUnix() {
+    return s_isUnix;
+  }
+
+  /** Check if the platform is Windows. */
+  public static boolean isWindows() {
+    return s_isWindows;
+  }
+
+  /**
+   * Try to open a URL in an external browser. Tries /usr/bin/open if Platform.isMac(), rundll32
+   * url.dll,FileProtocolHandler if Platform.isWindows(), and if isUnix() in this order: - xdg-open
+   * - kfmclient (if KDE is running) - firefox - mozilla - opera
+   *
+   * @param url URL to open.
+   * @return false if everything failed.
+   */
+  public static boolean openInExternalBrowser(URL url) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.